tests/mock_network/drop.rs

```diff
@@ -41,35 +43,12 @@ fn node_drops() {
         None,
     );
     let mut nodes = create_connected_nodes(&network, elder_size + 2);
-    drop_node(&mut nodes, 0);
 
-    // Trigger poll_and_resend to allow remaining nodes to gossip and
-    // update their chain accordingly.
-    poll_and_resend(&mut nodes);
-    verify_invariant_for_all_nodes(&network, &mut nodes);
-}
+    let dropped_node = drop_node(&mut nodes, 0);
```

Comment:
I think the `drop_node` function is now close to useless. What about removing it and doing just something like:
```rust
let dropped_nodes = iter::once(nodes.remove(0).id()).collect(); // collect the id of the removed node into a `BTreeSet`.
verify_dropped_nodes(&mut rng, &mut nodes, &dropped_nodes);
```

Reply:
I still think we shall keep that to confirm the disconnection won't cause status change immediately.

Reply:
Hm, OK. But maybe add a comment explaining why we are doing it? Something like we expect the node to remain elder until detected as unresponsive, or something like that.

Reply:
pushed updated comment